src/crate/client/sqlalchemy/tests/bulk_test.py

```diff
@@ -79,3 +109,62 @@ def test_bulk_save(self):
             ('Callisto', 37)
         )
         self.assertSequenceEqual(expected_bulk_args, bulk_args)
+
+    @skipIf(SA_VERSION < SA_2_0, "SQLAlchemy 1.x uses legacy bulk INSERT mode")
+    @patch('crate.client.connection.Cursor', FakeCursor)
+    def test_bulk_save_modern(self):
+        """
+        Verify modern SQLAlchemy bulk INSERT mode.
+
+        > A list of parameter dictionaries sent to the `Session.execute.params` parameter,
+        > separate from the Insert object itself, will invoke *bulk INSERT mode* for the
+        > statement, which essentially means the operation will optimize as much as
+        > possible for many rows.
+        >
+        > -- https://docs.sqlalchemy.org/orm/queryguide/dml.html#orm-queryguide-bulk-insert
+
+        > We have been looking into getting performance optimizations
+        > from `bulk_save()` to be inherently part of `add_all()`.
+        >
+        > -- https://github.com/sqlalchemy/sqlalchemy/discussions/6935#discussioncomment-1233465
+
+        > The remaining performance limitation, that the `cursor.executemany()` DBAPI method
+        > does not allow for rows to be fetched, is resolved for most backends by *foregoing*
+        > the use of `executemany()` and instead restructuring individual INSERT statements
+        > to each accommodate a large number of rows in a single statement that is invoked
+        > using `cursor.execute()`. This approach originates from the `psycopg2` fast execution
+        > helpers feature of the `psycopg2` DBAPI, which SQLAlchemy incrementally added more
+        > and more support towards in recent release series.
+        >
+        > -- https://docs.sqlalchemy.org/core/connections.html#engine-insertmanyvalues
+        """
+
+        # Adjust diff output on errors.
+        self.maxDiff = None
```

Comment:
What's the purpose of this, I don't get it, sorry.

Reply:
Thanks. I've amended the patch and rephrased this comment. Is it better?
```python
# Don't truncate unittest's diff output on `assertListEqual`.
self.maxDiff = None
```